app/src/main/java/com/team41/boromi/dbs/UserDB.java

```diff
@@ -0,0 +1,66 @@
+package com.team41.boromi.dbs;
+
+import static com.team41.boromi.constants.CommonConstants.DB_TIMEOUT;
+
+import android.util.Log;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.SetOptions;
+import com.google.gson.Gson;
+import com.team41.boromi.models.User;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class UserDB {
+
+  private final static String TAG = "UserDB";
+  private final String DB_COLLECTION = "users";
+  private final FirebaseFirestore db;
+  private final CollectionReference usersRef;
+
+  private final Gson gson = new Gson();
+
+  @Inject
+  public UserDB(FirebaseFirestore db) {
+    this.db = db;
+    usersRef = db.collection(DB_COLLECTION);
+  }
+
+  /**
+   * sync request for the user
+   *
+   * @param uuid
+   * @return throws runtimeException if failed, otherwise gives user
+   */
+  public User getUserByUUID(String uuid) {
+    DocumentSnapshot res;
+    try {
+      res = Tasks.await(usersRef.document(uuid).get(), DB_TIMEOUT, TimeUnit.MILLISECONDS);
+    } catch (Exception e) { // failed
+      Log.w(TAG, e.getCause());
+      return null;
+    }
+
+    if (res.exists()) {    // if user exists
+      return res.toObject(User.class);
+    } else {              // if user fails
+      return null;
+    }
+  }
+
+
+  /**
+   * async pushes a user to a db merges data only in case of conflicts
+   */
+  public void pushUser(User user) {
+    Gson gson = new Gson();
+    gson.toJson(user);
+    usersRef.document().set(user, SetOptions.merge());
```

Comment:
yes that's actually a good point, I didn't catch that! This would speed up uid lookups as well since we can just grab uids !

Reply:
actually if It's okay with you/others I would like to maintain that the user object knows about its UUID. Incase future implementation needs access to it. We can always remove the UUID as a field later on if it turns out to be useless :)

Reply:
That's fine, the important thing is the key to the user's document is the uid given to them during authentication. It looks like you've done that so I'll go ahead and approve the review so that we can move onto the demo